client/utils/moving_statistics/moving_average.c

```diff
@@ -0,0 +1,32 @@
+/***************************************************************************
+This software will calculate the moving average.
+Thanks to B.P.Welford (1962) [Donald Knuth's Art of Computer Programming, Vol 2, page 232, 3rd edition]
+
+Copyright (C) 2018  Marco Scarlino <marco.scarlino@students-live.uniroma2.it>.
+moving_average (consisting of moving_average.c and moving_average.h) is free software; 
+you can redistribute it and/or modify it.
+If you modify this file, you have to release the source code of this file included all of the modifications apported.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+***************************************************************************/
+
+#include "moving_average.h"
+
+
+void welford_one_pass(long double* mean, long double sample, uint16_t nsamples){
```

Comment:
Can you make the name of function more descriptive? welford_one_pass - it is a bit unclear what this function does.

Reply:
It's a famous algorithm, i'll add a small comment to explain